src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -598,4 +598,6 @@ async def wait_for_connection(self) -> None:
         return await self._call.wait_for_connection()
 
     async def read(self) -> ResponseType:
-        return await self._call.read()
+        # Behind the scenes everyting goes through the
+        # async iterator. So this path should not be reached.
+        raise Exception()
```

Comment:
Since an exception being raised here would indicate an error internal to the library, we should probably raise an `AssertionError`. Also, we probably want to move (something like) that comment to the error string so the user has a little more information to go on should they ever trigger this codepath somehow.